packages/react/src/components/List/List.tsx

```diff
@@ -614,18 +615,24 @@ export class List<T = any> extends React.Component<IListProps<T>, IListState<T>>
         itemKey = index;
       }
 
-      cells.push(
-        <div
-          role={cellRole}
-          className={'ms-List-cell'}
-          key={itemKey}
-          data-list-index={index}
-          data-automationid="ListCell"
-        >
-          {onRenderCell &&
-            onRenderCell(item, index, !this.props.ignoreScrollingState ? this.state.isScrolling : undefined)}
-        </div>,
-      );
+      const renderCell = onRenderCellConditional ? onRenderCellConditional : onRenderCell;
+
+      const cell =
+        renderCell?.(item, index, !this.props.ignoreScrollingState ? this.state.isScrolling : undefined) ?? null;
+
+      if (!onRenderCellConditional || (onRenderCellConditional! && cell !== null)) {
```

Comment:
Looking at it again I actually think the `onRenderCellConditional!` can entirely be dropped because the only way to get into that branch is for `onRenderCellConditional` to be defined so the first branch of the OR implies the truthiness in the second branch.

Changed to 
```ts
if (!onRenderCellConditional || cell) // ...
```